src/visualization/types/Graph/view.tsx

```diff
@@ -70,6 +78,15 @@ const XYPlot: FC<Props> = ({
   annotations,
   cellID,
 }) => {
+  const [resultState, setResultState] = useState(result)
+  const [preZoomResult, setPreZoomResult] = useState<InternalFromFluxResult>(
+    null
+  )
+
+  useEffect(() => {
+    setResultState(result)
+  }, [result])
```

Comment:
what's the difference between `resultState` and `result` here? I'm having trouble understanding the purpose of using `resultState`, when it seems like `result` is being changed to `resultState` lower down.

Reply:
`result` comes from the props and is responsible for the "normal" flow of the data. As the graph is updated (manually hitting the refresh button, for example), we want the graph's current state to match that and have the normal results.

`resultState` can also be affected by zooming in, which temporarily hijacks the graph and shows more fine grained detail. Hence, it is a local state for the view. If the user un-zooms, then we want to go back to the normal flow of data. Or if the user never zooms at all, we don't want to interfere with the `result`